launcher/src/main/java/org/apache/brooklyn/launcher/BrooklynLauncher.java

```diff
@@ -384,7 +384,7 @@ public void terminate() {
         if (getPersistMode() != PersistMode.DISABLED) {
             try {
                 Stopwatch stopwatch = Stopwatch.createStarted();
-                if (managementContext.getHighAvailabilityManager().getPersister() != null) {
+                if (managementContext.getHighAvailabilityManager() != null && managementContext.getHighAvailabilityManager().getPersister() != null) {
```

Comment:
Do you think we should do the same checks in `AbstractManagementContext.terminate()` in case startup failed?